core/src/main/scala/kafka/log/Log.scala

```diff
@@ -167,18 +172,18 @@ class Log(@volatile var dir: File,
       val filename = file.getName
       if(filename.endsWith(DeletedFileSuffix) || filename.endsWith(CleanedFileSuffix)) {
         // if the file ends in .deleted or .cleaned, delete it
-        file.delete()
+        Files.deleteIfExists(file.toPath)
```

Comment:
One thing that is not clear to me is if we want to just fail if we fail to delete or if we want to try to delete all the files and throw the exception at the end. What do you think?